### PR TITLE
Upgrade storybook version

### PR DESCRIPTION
**Pull request by negreirosleo**
#### What does this PR do?
It upgrades the storybook version to 5.3.18

#### Where should the reviewer start?
Anywhere, no particular order is required

#### What testing has been done on this PR?
I manually checked a couple of stories and looked for bugs

#### How should this be manually tested?
Just check the stories and assert that everything looks fine

#### Any background context you want to provide?
--

#### What are the relevant issues?
#3212

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
It is backwards compatible

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -72,10 +72,10 @@
     "@babel/preset-env": "^7.3.4",
     "@babel/preset-react": "^7.0.0",
     "@emotion/core": "^10.0.7",
-    "@storybook/addon-options": "~5.1.11",
-    "@storybook/addon-storysource": "~5.1.11",
-    "@storybook/addons": "~5.1.11",
-    "@storybook/react": "~5.1.11",
+    "@storybook/addon-options": "~5.3.18",
```
Comment (ShimiSun) [inline package.json:75]: would it work if we'll use the caret?
```suggestion
    "@storybook/addon-options": "^5.3.18",
```

Comment (negreirosleo) [reply package.json:75]: let me check

Comment (negreirosleo) [reply package.json:75]: I think theres no problem using `^`

Comment (negreirosleo) [reply package.json:75]: Unless storybook releases a breaking change out of the blue (which is very improbable), I see no issue, im updating this rn

**Comment by ShimiSun** [comment]
We are getting errors from https://www.chromaticqa.com/build?appId=5d9774839a6eff00203f5cbf&number=1785, can you take a look?

**Comment by negreirosleo** [comment]
@ShimiSun Looking into those erros rn

**Comment by negreirosleo** [comment]
@ShimiSun we should plan something for the files that arent stories but fall into the stories folder, cause we are now receiving console warns about it
image

**Comment by ShimiSun** [comment]
I agree. we should exclude them from the loader.

**Comment by negreirosleo** [comment]
@ShimiSun I address that problem in this PR or I leave it for the next one?

**Comment by ShimiSun** [comment]
@negreirosleo I think it can wait for the next PR.

File: package.json [hunk 2]
```diff
@@ -72,10 +72,10 @@
     "@babel/preset-env": "^7.3.4",
     "@babel/preset-react": "^7.0.0",
     "@emotion/core": "^10.0.7",
-    "@storybook/addon-options": "~5.1.11",
-    "@storybook/addon-storysource": "~5.1.11",
-    "@storybook/addons": "~5.1.11",
-    "@storybook/react": "~5.1.11",
+    "@storybook/addon-options": "^5.3.18",
+    "@storybook/addon-storysource": "^5.3.18",
+    "@storybook/addons": "^5.3.18",
```
Comment (ShimiSun) [inline package.json:77]: Can you check if we really need all those packages of addons? I think we need the story source one, not sure why we will need the others, see if we can do some package optimizations here.
Also, I think I read somewhere that story source might be deprecated and it is recommended to use another package, but I can't find it anywhere so I hope I'm not imagining anything.

Comment (negreirosleo) [reply package.json:77]: Sure!!

**Comment by negreirosleo** [comment]
@ShimiSun I removed storybook-options cause it was deprecated but storysource is being used :+1:

**Comment by ShimiSun** [comment]
Hi buddy, can you check why the Chromatic Chart examples are broken? example: https://deploy-preview-3978--sad-tereshkova-173d07.netlify.app/?path=/story/chart--zoom

**Comment by ShimiSun** [comment]
@negreirosleo any progress with this PR?

**Comment by negreirosleo** [comment]
@ShimiSun I'm having issues with the `data.js` through the stories, there are plenty of them, do I just move them one folder up?

**Comment by ShimiSun** [comment]
@negreirosleo Can we nest them on a folder under 'stories' and have that folder ignored by the loader? that will be best.

**Comment by negreirosleo** [comment]
@ShimiSun I totally forgot we could do that lol, thanks!

**Comment by ShimiSun** [comment]
Any updates on this PR @negreirosleo ?

**Comment by ShimiSun** [comment]
Can you please update the lock file?

**Comment by negreirosleo** [comment]
@ShimiSun I will, just noticed some issues with some chart stories I overlooked

File: storybook/preview.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
-import { addParameters, configure } from '@storybook/react';
+import { configure } from '@storybook/react';
```
Comment (ShimiSun) [inline storybook/preview.js:1]: where is the theme defined now that `addParameters` got removed?

File: src/js/components/Chart/stories/Scan.js [hunk 4]
```diff
@@ -123,5 +123,5 @@ const ScanChart = props => {
 };
 
 storiesOf('Chart', module).add('Scan', () => (
-  <ScanChart data={generateData(30, 100)} max={100} />
+  <ScanChart data={generateData()(30, 100)} max={100} />
```
Comment (ShimiSun) [inline src/js/components/Chart/stories/Scan.js:126]: I wondering if there is a simpler way to use generateData, it feels a little verbose.

File: storybook/preview.js (hunk 3 above)
Comment (ShimiSun) [inline storybook/preview.js:1]: Should you remove the `grommetLight` code if it's not being used?

Comment (negreirosleo) [reply storybook/preview.js:1]: That theme wasnt being used!
the only theme used on stories was this one: `src/js/all/stories/theme.js`

Comment (negreirosleo) [reply storybook/preview.js:1]: It was removed, wasnt it?

**Comment by negreirosleo** [comment]
@ShimiSun every file that was loaded by the storybook that isnt a story turns into this:
image

So in order to use the: `src/js/components/Chart/stories/data.js` and `src/js/all/stories/theme.js` I had to change their call.

On `src/js/all/stories/theme.js` I had to do:
```js
export const { colors } = hpe().global
```
because `hpe` was used as the decorated function that returned the object, so: `hpe().global`

On `src/js/components/Chart/stories/data.js` I had to create a function that returns a function because of this too.
```js
export const generateData = (count, max) => {} // Even if I call generateData(10,100) max will be undefined, because generateData turns into a decorated function.
export const generateData = () => (count,max) => {} // This way it generateData is the decorated function and generateData() returns the function that actually do what we want
```
To reduce verbosity I think we could make:
```js
export const decorator = () => (count,max) => {} // accepting name suggestions for this 
export const generateData = decorator()
```

and then call: 
```js
<ScanChart data={generateData(30, 100)} max={100} />
```

But i dont think its much better :thinking:

**Comment by ShimiSun** [comment]
> To reduce verbosity I think we could make:
> 
> export const decorator = () => (count,max) => {} // accepting name suggestions for this 
> export const generateData = decorator()
> and then call:
> 
> <ScanChart data={generateData(30, 100)} max={100} />
> But i dont think its much better 🤔

I actually think it will be cleaner this way.

**Comment by negreirosleo** [comment]
@ShimiSun Ok, fixing that to be cleaner!

**Comment by negreirosleo** [comment]
@ShimiSun the solution I suggested didnt work, but I made it cleaner by wraping the function call in an object, that way avoiding the creation of a decorated function

File: storybook/preview.js (hunk 3 above)
Comment (ShimiSun) [reply storybook/preview.js:1]: Nice clean up.

**Comment by ShimiSun** [comment]
I'm super excited about this PR, thank you @negreirosleo ❤️

**Pull request merged**